padl/dumptools/var2mod.py

```diff
@@ -596,8 +597,30 @@ def find_globals(node: ast.AST, filter_builtins: bool = True) -> Set[Tuple[str,
     return globals_
 
 
+def _check_and_make_increment(var: ScopedName, scope: Scope, scoped_name: ScopedName):
```

Comment:
I have a question. Do we need both `scope` and `scoped_name` here? I thought `scoped_name` also contains the `scope` of the target.

Reply:
Ahh I see further down how these can differ. Could you add to the `scope` doc string why this could differ from `scoped_name`?

Reply:
i moved scope to _check_and_make_increment, and added some comment to explain the things there.